metofficedatahub/multiple_files.py

```diff
@@ -207,119 +204,74 @@ def load_all_files(self) -> xr.Dataset:
+    Save dataset
+
+    :param dataset: The Xarray Dataset to be saved
+    :param save_dir: the directory where data is saved. This can either be a local path or a
+        "s3://..." path. 3 files similar files are created:
+            * One using the timestamp of the first `init_time` as the filename
+            * latest.netcdf
+            * latest.zarr
     """
 
-    logger.info(f"Saving data to {output_type} file here: {save_dir}")
-
-    assert output_type in ["zarr", "netcdf"]
+    logger.info(f'Saving data to "{save_dir}"')
 
-    # Make two files names
-    # 1. use the date timestamp of the data. Idea is that this will keep the historic
-    # 2. the latest - shows the most recent data without search through historic.
-    filename_and_path, filename_and_path_latest = make_output_filenames(
-        dataset=dataset, save_dir=save_dir, output_type=output_type
-    )
+    filename = _get_first_init_time_as_str(dataset)
+    path = f"{save_dir}/{filename}.netcdf"
+    logger.debug(f'Saving data to "{path}"')
+    save_to_s3(dataset, path)
 
-    # option to save latest or not
-    if save_latest:
-        logger.debug(f"Saving latest file {filename_and_path_latest}")
-        save_to_s3(
-            dataset=dataset, filename_and_path=filename_and_path_latest, output_type=output_type
-        )
+    # Also save it as "lastest.<ext>", both in zarr and netcdf format.
+    # TODO Copying the file we just wrote in AWS directly would be faster.
+    logger.debug(f'Saving data to "{path}"')
+    path = f"{save_dir}/latest.netcdf"
+    save_to_s3(dataset, path)
 
-    # save historic data
-    logger.debug(f"Saving file {filename_and_path}")
-    save_to_s3(dataset=dataset, filename_and_path=filename_and_path, output_type=output_type)
+    chunked = _chunk(dataset, ideal_chunk_size)
+    logger.debug(f'Saving data to "{path}"')
+    path = f"{save_dir}/latest.zarr"
+    save_to_s3(chunked, path)
 
 
-def save_to_s3(dataset: xr.Dataset, filename_and_path: str, output_type):
+def save_to_s3(dataset: xr.Dataset, path: str):
     """Save to s3"""
-
-    if output_type == "zarr":
-        # encoding used when saving to zarr file
-        encoding = {
-            var: {"compressor": numcodecs.Blosc(cname="zstd", clevel=5)}
-            for var in dataset.data_vars
-        }
-
-        dataset.to_zarr(store=filename_and_path, mode="w", encoding=encoding, consolidated=True)
+    if path.endswith(".zarr"):
+        dataset.to_zarr(store=path, mode="w", consolidated=True)
+    elif path.endswith(".netcdf"):
+        # xarray doesn't support writing .netcdf files directly to S3 like for .zarr files.
+        # Also note the "simplecache::" and see https://github.com/pydata/xarray/issues/4122
+        with fsspec.open("simplecache::" + path, mode="wb") as f:
+            dataset.to_netcdf(f)
     else:
-        save_to_netcdf_to_s3(dataset=dataset, filename=filename_and_path)
-
-
-def save_to_netcdf_to_s3(dataset: xr.Dataset, filename: str):
```

Comment:
Thanks for adding the note. I guess we will find out quite quickly if this doesnt work, i.e doesnt save to s3